app/models/katello/sync_plan.rb

```diff
@@ -23,13 +23,20 @@ class SyncPlan < Katello::Model
     validates :interval, :inclusion => {:in => TYPES}, :allow_blank => false
     validates :enabled, :inclusion => [true, false]
     validate :validate_sync_date
+    validate :product_enabled
     validates_with Validators::KatelloNameFormatValidator, :attributes => :name
 
     scoped_search :on => :name, :complete_value => true
     scoped_search :on => :organization_id, :complete_value => true, :only_explicit => true, :validator => ScopedSearch::Validators::INTEGER
     scoped_search :on => :interval, :complete_value => true
     scoped_search :on => :enabled, :complete_value => true
 
+    def product_enabled
+      products.each do |product|
+        errors.add :product, _("#{product.name} can not be disabled.") unless product.enabled?
```

Comment:
Looking good -- only thing I would update is that in the context of adding to a sync plan this message comes off as confusing. How about "Can not add disabled product Blah Blah Blah" or "Can not add product Blah Blah because it is disabled"